### ISSUE TITLE
Backend request processing has high overhead on macOS

### ISSUE THREAD

**Issue by aturon** [fastly/Viceroy#101]
A simple proxy service that forwards a request upstream adds significant overhead on macOS, but not on Linux.

Can be reproduced by trying:

```
curl -s http://localhost:7676/ -o /dev/null -w %{time_connect}:%{time_starttransfer}:%{time_total}
```

against a Viceroy forwarding service, vs the same against the backend URL directly. On Linux, the two reliably produce similar timings, while on macOS the Viceroy service takes 2-3x as long as going directly to the backend.

**Comment by aturon** [comment]
Good news! This actually appears to be fixed by #100. cc @williamoverton

**Comment by aturon** [comment]
For posterity: one thing that #100 changes is that we load up certificates once, at the beginning of execution, rather than every time we connect to a backend (which was implicitly happening by allocating a hyper_rustls connection). That, unsurprisingly, was adding quite a bit of overhead.
